test/unit_tests/utilities/check_return.hpp

```diff
@@ -18,7 +18,7 @@
 #include <string>
 
 // Check function for non-zero return flag
-static int check_flag(int flag, const std::string funcname)
+inline int check_flag(int flag, const std::string funcname)
```

Comment:
While you're updating this
```suggestion
inline int check_flag(int flag, const std::string& funcname)
```